scripts/pi-hole/php/add.php

```
@@ -18,47 +18,90 @@
 // filters are validated by FTL on import and skipped if invalid
-if($type !== "regex") {
+$check_lists = ["white","black","audit"];
+if(in_array($list, $check_lists)) {
     check_domain();
 }
 
 // Escape shell metacharacters
-$domains = escapeshellcmd($_POST['domain']);
+$domains = explode(",",$_POST['domain']);
 
-switch($type) {
-    case "white":
-        if(!isset($_POST["auditlog"]))
-            echo shell_exec("sudo pihole -w --web ".$domains);
-        else
-        {
-            echo shell_exec("sudo pihole -w --web ".$domains);
-            echo shell_exec("sudo pihole -a audit ".$domains);
-        }
-        break;
-    case "black":
-        if(!isset($_POST["auditlog"]))
-            echo shell_exec("sudo pihole -b --web ".$domains);
-        else
-        {
-            echo shell_exec("sudo pihole -b --web ".$domains);
-            echo shell_exec("sudo pihole -a audit ".$domains);
-        }
-        break;
-    case "black_regex":
-        echo shell_exec("sudo pihole --regex --web ".$domains);
-        break;
-    case "white_regex":
-        echo shell_exec("sudo pihole --whiteregex --web ".$domains);
-        break;
-    case "black_wild":
-        echo shell_exec("sudo pihole --wild --web ".$domains);
-        break;
-    case "white_wild":
-        echo shell_exec("sudo pihole --whitewild --web ".$domains);
-        break;
-    case "audit":
-        echo shell_exec("sudo pihole -a audit ".$domains);
-        break;
+require_once("func.php");
+
+require("database.php");
+$GRAVITYDB = getGravityDBFilename();
+$db = SQLite3_connect($GRAVITYDB, SQLITE3_OPEN_READWRITE);
+
+function add_to_table($table, $domains, $wildcardstyle=false)
+{
+	global $db;
+	// Prepare SQLite statememt
+	$stmt = $db->prepare("INSERT OR IGNORE INTO ".$table." (domain) VALUES (:domain);");
+
+	// Return early if we prepare the SQLite statement
+	if(!$stmt)
+	{
+		echo "Failed to prepare statement for ".$table." table.";
+		echo $sql;
+		return 0;
```

This should return the error string like the rest of the function.